Watchman.Web/Areas/LogsViewer/Services/LogsService.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Serilog;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+using Watchman.Web.Areas.LogsViewer.Models.Dtos;
+
+namespace Watchman.Web.Areas.LogsViewer.Services
+{
+    public class LogsService
+    {
+        public IEnumerable<LogDto> GetLogs(GetLogsRequest request)
+        {
+            foreach (var filePath in Directory.GetFiles("logs").Where(x => Path.GetExtension(x) == ".json"))
+            {
+                using var stream = File.Open(filePath, FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
+                using var streamReader = new StreamReader(stream);
+
+                var content = streamReader.ReadToEnd();
+                var normalizedContent = "[" + content.Trim(",[]".ToCharArray()) + "]";
+                var logs = JsonConvert.DeserializeObject<IEnumerable<LogDto>>(normalizedContent);
+
+                foreach (var log in logs.Where(x => this.ShouldReturn(request, x)))
+                {
+                    yield return log;
+                }
```

Comment:
dlaczego nie `return logs.Where(x => this.ShouldReturn(request, x))` ?

Reply:
Bo tam wyżej jest jeszcze foreach 
musielibyśmy sie bawić w tworzenie zmiennej z listą list i zwracać `SelectMany(x => x)`
a tak to mamy prostego foreach z yieldem